24.md

```diff
@@ -0,0 +1,497 @@
+
+Once `Kind:14` is created, remove the signature and serialize as JSON. Encrypt the serialized `Kind:14` to the receiver's public key and seal it with a `kind:13` event, signed by the sender's main key. 
+
+For each receiver: 
+
+1. Create a new key, serialize the seal, and encrypt it into a `kind:1059` with a tag to the receiver, signed by the new random key. 
+2. Broadcast the `kind:1059` event to the relays of the receiver. 
+
+For the sender: 
+
+1. Create a new key, serialize the seal, and encrypt it into a `kind:1059` with a tag to the sender, signed by the new random key. 
+2. Broadcast the `kind:1059` event to the relays of the sender. 
+
+## 6. Private Group Chat Algorithm
+
+Private Group chats are `Kind:14`s for more than one `p` tag. There is no channel definition event. The group of authors + p' tags is a group. 
+
+To send a private message to a group, create a kind `Kind:14` event with the message in its `.content` and add a p-Tag to each receiver. Other tags are encouraged to be added, such as `content-sensitivity` if the content is NSFW and `subject` to create a name to the conversation. Once `Kind:14` is created, remove the signature and serialize it as JSON.
+
+For each receiver: 
+
+1. Encrypt the serialized `Kind:14` to the **receiver's public key** and seal it with a `kind:13` event, signed by the sender's main key. 
+2. Create a new key, serialize the seal, and encrypt it into a `kind:1059` with a tag to the **receiver**, signed by the new random key. 
+3. Broadcast the `kind:1059` event to the relays of the **receiver**. 
+
+For the sender: 
+
+1. Create a new key, serialize the seal, and encrypt it into a `kind:1059` with a tag to the **sender**, signed by the new random key. 
+2. Broadcast the `kind:1059` event to the relays of the **sender**. 
+
+Clients SHOULD render every different group of (author + p' tags) as a group with continuing conversations. If a new p tag is added, a new group is created. 
+
+The `subject` tag defines any name for the group or the current topic of the discussion. Any member of the group can change the subject. 
+
+## 7. Final Considerations
+
+This proposal makes heavy use of one-time keys to elevate the privacy design of Nostr messages. It can be applied to any Nostr event kind (e.g. private reactions), but the focus here is on private messages. For instance, the use of a Gossip Event enhances a paid-subscriber-only event kind by making it difficult (not impossible) to leak the paid content in a publicly verifiable payload. 
+
+It's worth noting that the only created_at time that matters is with the Gossip event. All the other time-attributes can be tweaked to reduce the time-collision of send and receive Gift Wraps and Sealed Events.
```

Comment:
For the purposes of balancing retrieval and privacy, maybe we could recommend how much to randomize the timestamp. Now +/- one day? One week?

Reply:
```suggestion
It's worth noting that the only created_at time that matters is with the Gossip event. All the other time-attributes SHOULD be tweaked to reduce the time-collision of send and receive Gift Wraps and Sealed Events.

Relays may choose not to store gift wrapped events due to them not being publicly useful. Clients MAY choose to attach a certain amount of proof-of-work to the wrapper event per NIP-13 in a bid to demonstrate that the event is not spam or a denial-of-service attack.

To protect recipient metadata, relays SHOULD only serve kind 1059 events intended for the marked recipient. When possible, clients should only send wrapped events to read relays for the recipient that implement AUTH, and refuses to serve wrapped events to non-recipients.
```